sos/report/plugins/system.py

```diff
@@ -33,5 +33,11 @@ def setup(self):
             "/proc/sys/net/ipv6/neigh/*/base_reachable_time"
         ])
 
+        # collect glibc tuning decisions
+        self.add_cmd_output([
+            "/usr/bin/ld.so --help",
+            "/usr/bin/ld.so --list-diagnostics"
```

Comment:
Should we also be collecting `ld.so --list-tunables`?

Reply:
Nothing like that has been asked for.

Reply:
We should probably clarify - the commit and comment both reference tuning, but doesn't collect the only thing in the help output that references tuning:

```
  --list-tunables       list all tunables with minimum and maximum values
```

Reply:
I wouldn't mind adding that, too, but the minimums/maximums currently aren't dynamic, so they won't confer additional system-specific information.

Cc @siddhesh

Reply:
mins/maxes might not, but surely the actual value would?

```
# ld.so --list-tunables
glibc.rtld.nns: 0x4 (min: 0x1, max: 0x10)
glibc.elision.skip_lock_after_retries: 3 (min: 0, max: 2147483647)
glibc.malloc.trim_threshold: 0x0 (min: 0x0, max: 0xffffffffffffffff)
```

Reply:
>> I wouldn't mind adding that, too, but the minimums/maximums currently aren't dynamic, so they won't confer additional system-specific information.

> mins/maxes might not, but surely the actual value would?

The x86-specific tunables (`x86_data_cache_size` for example) will depend on the target CPU since it gets the default value from `_SC_LEVEL1_DCACHE_SIZE`, so it could return different values on different machines.  `glibc.malloc.trim_threshold` though will just return 0 if not set since the default is dynamic, i.e. it starts at 128K and the malloc implementation changes it at runtime.  A single value is misleading since it implies that it is the single default value.  Many tunables however should have a constant default value.

So in summary, "it depends".

Reply:
Okay, then let's include it, too.